frontend/src/components/cards/IconSwitcher.js

```diff
@@ -0,0 +1,31 @@
+import React, { useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import LandscapeIcon from './avatarIcons/LandscapeIcon';
+import ArchitectureIcon from './avatarIcons/ArchitectureIcon';
+import HarbourIcon from './avatarIcons/HarbourIcon';
+import NightIcon from './avatarIcons/NightIcon';
+import ImageIcon from './avatarIcons/ImageIcon';
+import PlacesContext from '../../contexts/PlacesContext';
+
+export default function IconSwitcher() {
+  const { id } = useParams();
+  const { places } = useContext(PlacesContext);
+  const placeData = places.find((place) => place.id === id);
```

Comment:
Overall, this is way too complicated. Why not simply a function that get's the type of your place and returns the icon?

Reply:
my switch was intended to be such a simple function. :-(

Reply:
i wrote my switch in iconswitcher.js without breaks now, but this leaves me with a TypeError: Cannot read property 'type' of undefined and i am stuck again